src/test/java/com/microsoft/jenkins/kubernetes/integration/integration-test.sh

```diff
@@ -0,0 +1,193 @@
+        -h|--master-host)
+            export KUBERNETES_CD_MASTER_HOST="$2"
+            shift; shift
+            ;;
+        -u|--admin-user)
+            export KUBERNETES_CD_ADMIN_USER="$2"
+            shift; shift
+            ;;
+        -k|--ssh-key-file)
+            export KUBERNETES_CD_KEY_PATH="$2"
+            shift; shift
+            ;;
+        --docker-registry)
+            export KUBERNETES_CD_DOCKER_REGISTRY="$2"
+            shift; shift
+            ;;
+        --docker-username)
+            export KUBERNETES_CD_DOCKER_USERNAME="$2"
+            shift; shift
+            ;;
+        --docker-password)
+            export KUBERNETES_CD_DOCKER_PASSWORD="$2"
+            shift; shift
+            ;;
+        --docker-repository)
+            export KUBERNETES_CD_DOCKER_REPOSITORY="$2"
+            shift; shift
+            ;;
+        --skip-clean)
+            export SKIP_CLEAN=true
+            shift
+            ;;
+        *)
+            show_help
+            ;;
+    esac
+done
+
+# common suffix for the names
+suffix=$(xxd -p -l 4 /dev/urandom)
+
+resource_group=
+if [[ -z "$KUBERNETES_CD_MASTER_HOST" ]] || [[ -z "$KUBERNETES_CD_DOCKER_REPOSITORY" ]]; then
+    resource_group="k8s-test-$suffix"
+fi
+
+export integration_test_script_pid=$$
+export temp_config_dir=kubeconfig.tmp
+
+post_clean_up() {
+    set +x
+    # wait the testing process to finish
+    while ps -p "$integration_test_script_pid" >/dev/null; do
+        sleep 0.5
+    done
+    set -x
+
+    if [[ -d "$temp_config_dir" ]]; then
+        echo "Clean up temporary directory $temp_config_dir"
+        rm -rf -- "$temp_config_dir"
+    fi
+
+    if [[ -n "$resource_group" ]]; then
+        echo "Clean up resource group $resource_group"
+        az group delete --yes --no-wait --name "$resource_group"
+    fi
+}
+
+if [[ -n "$resource_group" ]]; then
+    az group create --name "$resource_group" --location SoutheastAsia
+
+    if [[ "$SKIP_CLEAN" != "true" ]]; then
+        echo "Registering the clean up process..."
+        post_clean_up & disown
+    fi
+
+    if [[ -z "$KUBERNETES_CD_MASTER_HOST" ]]; then
+        k8s_name="k8s-$suffix"
+        az acs create --orchestrator-type kubernetes --resource-group "$resource_group" --name "$k8s_name" --agent-count 2 &
+        k8s_pid=$!
+    fi
+
+    if [[ -z "$KUBERNETES_CD_DOCKER_REPOSITORY" ]]; then
+        acr_name="acr$suffix"
+        az acr create --resource-group "$resource_group" --name "$acr_name" --sku Basic --admin-enabled true
+        export KUBERNETES_CD_DOCKER_REPOSITORY="$(az acr show --resource-group "$resource_group" --name "$acr_name" --query "loginServer" --output tsv)"
+        export KUBERNETES_CD_DOCKER_USERNAME="$acr_name"
+        export KUBERNETES_CD_DOCKER_PASSWORD="$(az acr credential show --name "$acr_name" --query "passwords[0].value" --output tsv)"
+        export KUBERNETES_CD_DOCKER_REGISTRY="http://$KUBERNETES_CD_DOCKER_REPOSITORY"
+
+        docker pull nginx
```

Comment:
This should be called even if KUBERNETES_CD_DOCKER_REPOSITORY is empty?

Reply:
The intention here is:

* if we test against existing docker repository, we pass in all the details of the repo and then we skip the repository setup
* otherwise if `KUBERNETES_CD_DOCKER_REPOSITORY` is empty, we setup new ACR instance and prepare the image for testing